client/active-game/active-game-manager.js

```diff
@@ -50,15 +51,28 @@ export default class ActiveGameManager extends EventEmitter {
         // Same config as before, no operation necessary
         return current.id
       }
-      // Quit the currently active game so we can replace it
-      this.emit('gameCommand', current.id, 'quit')
+      if (current.id !== config.gameId) {
+        // Quit the currently active game so we can replace it
+        this.emit('gameCommand', current.id, 'quit')
+      }
     }
     if (!config.setup) {
       this._setStatus(GAME_STATUS_UNKNOWN)
       this.activeGame = null
       return null
     }
 
+    if (current && current.routes && config.setup) {
+      const routesIds = new Set(current.routes.map(r => r.for))
+      const slotIds = new Set(config.setup.slots.map(s => s.id))
+
+      if (!slotIds.isSuperset(routesIds)) {
```

Comment:
Feels like we should probably be logging something in this case, or this will require some work to track down "why the game status suddenly became `UNKNOWN`" when this error is encountered. I also think throwing an exception would potentially be fine, given that this is a programmer error